.github/workflows/main.yml

```
@@ -0,0 +1,53 @@
+name: Build
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
```

```suggestion
on:
  release:
    types: [published]
```

Create the binaries only when release created